newlib/libc/sys/vita/select.c

```diff
@@ -0,0 +1,143 @@
+/*
+
+Copyright (C) 2016, Sunguk Lee
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+*/
+#include <sys/select.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <psp2/kernel/threadmgr.h>
+#include <psp2/net/net.h>
+#include <psp2/types.h>
+
+#include "vitadescriptor.h"
+
+#define SCE_ERRNO_MASK 0xFF
+
+#define MAX_EVENTS 255
+
+#define LOCK_ARRAY_LENGTH   (MAX_OPEN_FILES / 8) + ((MAX_OPEN_FILES % 8) ? 1 : 0)
+#define BIT_SET(arr, idx)	(arr)[(idx)/8] |= (1<<((idx)%8))
+#define IS_SET(arr, idx)	((arr)[(idx)/8] & (1<<((idx)%8))) != 0
```

Comment:
i think cannot direct use FD_SET.
FD_SET limitation is these macros only allow fd_set type.
and newlibs define's FD_SETSIZE is 64 not 128.